esm_parser/esm_parser.py

```diff
@@ -2198,19 +2198,26 @@ def could_be_int(value):
 def could_be_float(value):
     try:
         float(value)
-        return True
+        return contains_underscore(value)
     except:
         return False
 
 
 def could_be_complex(value):
     try:
         complex(value)
-        return True
+        return contains_underscore(value)
     except:
         return False
 
 
+def contains_underscore(value):
+    if isinstance(value, str):
+        if "_" in value:
+            return False
```

Comment:
is the underscore special in any way? why not test for all non-numeric characters?

Reply:
In principle non-numeric characters will not pass the evaluation in `could_be_int` method or in`could_be_float`, (i.e. `try int(value)`), so underscores seems to me like an exception, because `int("0000_000")` makes it to `0`.

Reply:
@Deniz, you are right, I will change the name of the function to something like `does_not_contain_underscore` cause I want to get `False` if it contains an underscore.